src/main/java/io/debezium/connector/vitess/VitessConnector.java

```diff
@@ -220,17 +226,32 @@ else if (gtidsPerShard != null && !currentShards.containsAll(gtidsPerShard.keySe
             if (maxTasks > 1) {
                 throw new IllegalArgumentException("Only a single connector task may be started");
             }
-            final List<String> gtids = connectorConfig.getGtid();
-            if (connectorConfig.getShard() != null &&
-                    gtids != VitessConnectorConfig.DEFAULT_GTID_LIST &&
-                    gtids != VitessConnectorConfig.EMPTY_GTID_LIST &&
-                    connectorConfig.getShard().size() != connectorConfig.getGtid().size()) {
-                throw new IllegalArgumentException("If GTIDs are specified must be specified for all shards");
-            }
+            verifyShardGtidConfig();
             return Collections.singletonList(properties);
         }
     }
 
+    private Map<String, String> getGtidsPerShardFromConfig() {
+        Map<String, String> gtidsPerShard = null;
+        if (connectorConfig.getShard() != null && connectorConfig.getGtid().size() > 1) {
+            gtidsPerShard = IntStream.range(0, connectorConfig.getShard().size())
+                    .boxed()
+                    .collect(Collectors.toMap(connectorConfig.getShard()::get, connectorConfig.getGtid()::get));
+            LOGGER.info("Found GTIDs per shard in config {}", gtidsPerShard);
+        }
+        return gtidsPerShard;
+    }
+
+    private void verifyShardGtidConfig() {
+        final List<String> gtids = connectorConfig.getGtid();
+        if (connectorConfig.getShard() != null &&
```

Comment:
This is re-used check - maybe it would be good to extract it it to a separate method at `VitessConnectorConfig`

Reply:
Added, thanks!